git_machete/generated_docs.py

```diff
@@ -0,0 +1,938 @@
+from typing import Dict
+
+# ---------------------------------------------------------------------------------------------------------
+# Warning: This file is NOT supposed to be edited directly, but instead regenerated via tox -e docs
+# ---------------------------------------------------------------------------------------------------------
+
+long_docs: Dict[str, str] = {
+    "add": """
+        <b>Usage:</b><b>
+          git machete add [-o|--onto=<target-upstream-branch>] [-R|--as-root] [-y|--yes] [<branch>]</b>
+
+        Adds the provided <branch> (or the current branch, if none specified) to the definition file.
+        If <branch> is provided but no local branch with the given name exists:
+        * if a remote branch of the same name exists in exactly one remote, then user is asked whether to check out this branch locally (as in `git checkout`),
+        * otherwise, user is asked whether it should be created as a new local branch.
+
+        If the definition file is empty or `-R/--as-root` is provided, the branch will be added as a root of the tree of branch dependencies.
+        Otherwise, the desired upstream (parent) branch can be specified with `-o/--onto`.
+        Neither of these options is mandatory, however; if both are skipped, git machete will try to automatically infer the target upstream.
+        If the upstream branch can be inferred, the user will be presented with inferred branch and asked to confirm.
+
+        Note: all the effects of `add` (except git branch creation) can as well be achieved by manually editing the definition file.
+
+        <b>Options:</b>
+           <b>-o</b>, <b>--onto=<target-upstream-branch></b>
+              Specifies the target parent branch to add the given branch onto. Cannot be specified together with `-R/--as-root`.
+           <b>-R</b>, <b>--as-root</b>
+              Add the given branch as a new root (and not onto any other branch). Cannot be specified together with `-o/--onto`.
+           <b>-y</b>, <b>--yes</b>
+              Don't ask for confirmation whether to create the branch or whether to add onto the inferred upstream.
+   """,
+    "advance": """
+        <b>Usage:</b><b>
+          git machete advance [-y|--yes]</b>
+
+        Fast forwards (as in `git merge --ff-only`) the current branch `C` to match its downstream `D`, pushes `C`
+        and subsequently slides out `D`. All three steps require manual confirmation unless `-y/--yes` is provided.
+
+        The downstream `D` is selected according to the following criteria:
+        * if `C` has exactly one downstream (child) branch `d` connected with a [32mgreen edge[0m (see status for definition) to `C` or is overridden, then `d` is selected as `D`,
```

Comment:
Hmmm there's a major inconsistency: while bold is kept in this file as `<b>` tags, colors are rendered directly with ANSI escape codes. Is there a strong reason for that? If not, I'd make it all consistent, TBD in which direction (all ANSI codes or no ANSI codes)

Also... what if the help is printed out to a non-terminal (see https://github.com/VirtusLab/git-machete/issues/595)? do ANSI codes for colors stay as they are? 🤔